sites/all/modules/ding_unilogin/ding_unilogin.admin.inc

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * @file
+ * Implements the administration UI for the module.
+ */
+
+/**
+ * UNI•Login settings form.
+ */
+function ding_unilogin_admin_settings_form($form, &$form_state) {
+  // Settings for the basic login service.
+  $form['login'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('UNI•Login'),
+    '#description' => t('Settings for the base login service.'),
+  );
+
+  $form['login']['ding_unilogin_id'] = array(
+    '#type' => 'textfield',
+    '#title' => t('UNI•Login ID'),
+    '#description' => t('The UNI•Login ID. Leave empty to disable UNI•Login.'),
+    '#default_value' => variable_get('ding_unilogin_id', ''),
+  );
+
+  $form['login']['ding_unilogin_secret'] = array(
+    '#type' => 'textfield',
+    '#title' => t('UNI•Login shared secret'),
+    '#description' => t('The shared secret used to calculate hashes.'),
+    '#default_value' => variable_get('ding_unilogin_secret', ''),
+  );
+
+  $form['login']['ding_unilogin_client_id'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Publizon client ID for UNI•Login users'),
```

Comment:
Is the reference to publizon necessary? If the module has potential to become a generic UNI-C login module for Ding2 then having a reference to a specific provider is unfortunate.